src/twisted/application/runner/test/test_pidfile.py

```diff
@@ -31,7 +30,7 @@
 from twisted.trial.unittest import SkipTest
 
 
-def ifPlatformSupported(f):
+def ifPlatformSupported(f: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
You could make the return type of the `Callable` stricter using something like this:
```py
TestMethodRT = TypeVar("TestMethodRT", None, defer.Deferred)
```
I'm not sure it's worth the effort though.

Reply:
This is a test module, no it's not public, and this is just to select the tests to run so… seems not worth it.  My pedantic self is tempted, but they it's mad that `Deferred` can't yet be stated as `Deferred[Any]` and so cannot be satisfied.